associations/serializers/events.py

```diff
@@ -0,0 +1,39 @@
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from associations.models import User, Event
+
+
+class EventSerializer(serializers.ModelSerializer):
+    participants = serializers.PrimaryKeyRelatedField(queryset=User.objects.all(), many=True, read_only=False)
+
+    class Meta:
+        model = Event
+        read_only_fields = ('id', 'association')
+        fields = read_only_fields + ('name', 'description', 'participants', 'starts_at', 'ends_at', 'place')
+
+    def is_valid(self, raise_exception=False):
+        """Check if the dates are consistent."""
+
+        self._errors = {}
+
+        if 'starts_at' in self.initial_data and 'ends_at' in self.initial_data:
+            if self.initial_data['starts_at'] >= self.initial_data['ends_at']:
+                self._errors = {'field_errors': 'field starts_at is not consistent with field ends_at.'}
+
+        if self._errors and raise_exception:
+            raise ValidationError(self._errors)
+
+        return super(EventSerializer, self).is_valid(raise_exception)
+
+    def update(self, instance, validated_data):
```

Comment:
This function is a bit redundant with `is_valid`. I think that `is_valid` must be called before `update` for serializers

Reply:
Unfortunately, `is_valid` does not give access to the updated instance. So we either do that verification in `update` or in the view. I think both are the same (but at least, here, we are being provided the instance).